specification/http/1.0/openapi.json

```diff
@@ -0,0 +1,456 @@
+                      "available_modes": {
+                        "contains": { "const": "value" }
+                      }
+                    }
+                  },
+                  "then": {
+                    "required": ["value"]
+                  }
+                }
+              ],
+              "additionalProperties": false
+            },
+            "requirements": {
+              "type": "object",
+              "properties": {
+                "authorization": {
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "id": {
+                        "type": "string",
+                        "description": "A provider's unique identifier, allowing the tool to specify a specific authorization provider."
+                      },
+                      "oauth2": {
+                        "type": "object",
+                        "description": "OAuth 2.0-specific authorization details.",
+                        "properties": {
+                          "scopes": {
+                            "type": "array",
+                            "items": {
+                              "type": "string"
+                            }
+                          }
+                        },
+                        "additionalProperties": false
+                      }
+                    }
+                  },
+                  "required": ["id"],
+                  "additionalProperties": false
+                },
+                "secrets": {
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "id": {
+                        "type": "string",
+                        "description": "The secret's unique identifier."
+                      }
+                    },
+                    "required": ["id"],
+                    "additionalProperties": false
+                  }
+                }
+              },
+              "additionalProperties": false
+            }
+          },
+          "required": ["id", "name", "description", "toolkit"],
+          "additionalProperties": false
+        },
+        "ExecuteToolRequest": {
+          "type": "object",
+          "properties": {
+            "execution_id": {
+              "type": "string",
+              "description": "The globally-unique ID for this tool execution."
+            },
+            "trace_id": {
+              "type": "string",
+              "description": "A trace identifier for the tool execution."
+            },
+            "tool_id": {
+              "$ref": "#/components/schemas/ToolId"
+            },
+            "input": {
+              "type": "object",
+              "description": "The input parameters for the tool call.",
+              "additionalProperties": true
+            },
+            "context": {
```

Comment:
Could you explain why the authorization information is needed here? Is the idea that for some tools the only way to interact w/ them is to just pass API keys to them directly? (i.e., no oauth flow)